tokenizers/src/processors/sequence.rs

```diff
@@ -0,0 +1,138 @@
+use crate::processors::PostProcessorWrapper;
+use crate::tokenizer::{Encoding, PostProcessor, Result};
+use crate::utils::macro_rules_attribute;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq)]
+#[macro_rules_attribute(impl_serde_type!)]
+pub struct Sequence {
+    processors: Vec<PostProcessorWrapper>,
+}
+
+impl Sequence {
+    pub fn new(processors: Vec<PostProcessorWrapper>) -> Self {
+        Self { processors }
+    }
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum SequenceProcessorError {
+    #[error("could not pop encodings vector")]
+    EncodingsVecPop,
+}
+
+impl PostProcessor for Sequence {
+    fn added_tokens(&self, is_pair: bool) -> usize {
+        self.processors
+            .iter()
+            .map(|p| p.added_tokens(is_pair))
+            .sum::<usize>()
+    }
+
+    fn process(
+        &self,
+        encoding: Encoding,
+        pair_encoding: Option<Encoding>,
+        add_special_tokens: bool,
+    ) -> Result<Encoding> {
+        let mut encodings = vec![encoding];
+        if let Some(encoding) = pair_encoding {
+            encodings.push(encoding);
+        }
+
+        for processor in &self.processors {
+            encodings = processor.process_chain(encodings, add_special_tokens)?;
+        }
+
+        match encodings.len() {
+            1 => Ok(encodings
+                .pop()
+                .ok_or(SequenceProcessorError::EncodingsVecPop)?),
+            _ => {
+                // merge encodings
+                for (i, encoding) in encodings.iter_mut().enumerate() {
+                    encoding.set_sequence_id(i);
+                }
+                Ok(Encoding::merge(encodings, false))
```

Comment:
to emulate https://github.com/huggingface/tokenizers/blob/bb6fea033932ff1fb23807a2b7c5376f5a7c4786/tokenizers/src/tokenizer/mod.rs#L123-L125